fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -242,13 +242,30 @@ public void testLimitWithContinuation() {
                     .setScore(0.0024906613F)
                     .setShard(0)
                     .build();
-            assertEquals(48, recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, IndexScanType.BY_LUCENE,
+            assertEquals(48, recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, IndexScanType.BY_LUCENE_FULL_TEXT,
                     TupleRange.allOf(Tuple.from("idiot")), continuation.toByteArray(),
                     ExecuteProperties.newBuilder().setReturnedRowLimit(50).build().asScanProperties(false))
                     .getCount().join());
         }
     }
 
+    @Test
+    public void testLimitNeedsMultipleScans() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, metaDataBuilder -> {
+                metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+                metaDataBuilder.addIndex(SIMPLE_DOC, SIMPLE_TEXT_SUFFIXES);
+            });
+            for (int i = 0; i < 800; i++) {
+                recordStore.saveRecord(createSimpleDocument(1623L + i, DYLAN, 2));
+            }
+            assertEquals(251, recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, IndexScanType.BY_LUCENE_FULL_TEXT,
+                    TupleRange.allOf(Tuple.from("idiot")), null,
+                    ExecuteProperties.newBuilder().setReturnedRowLimit(251).build().asScanProperties(false))
+                    .getCount().join());
+        }
```

Comment:
There should be some instrumentation in `performSearch` and this should assert on the number of times that was done.